src/main/scala/com/amazon/deequ/checks/Check.scala

```diff
@@ -368,37 +368,42 @@ case class Check(
    *   val check = new Check(CheckLevel.Error, "Data Synchronization Check")
-   *     .isDataSynchronized(otherDataFrame, columnMappings, assertionFunction)
+   *     .isDatasetMatched(otherDataFrame, columnMappings, assertionFunction)
    *
    *   val verificationResult = VerificationSuite()
    *     .onData(baseDataFrame)
    *     .addCheck(check)
    *     .run()
    * }}}
    *
-   * This will add a data synchronization check to the VerificationSuite, comparing the specified columns of
+   * This will add a dataset match check to the VerificationSuite, comparing the specified columns of
    * baseDataFrame and otherDataFrame based on the provided assertion function.
    *
-   *
    * @param otherDf         The DataFrame to be compared with the current one. Analyzed in conjunction with the
    *                        current DataFrame to assess data synchronization.
-   * @param columnMappings  A map defining the column correlations between the current DataFrame and otherDf.
+   * @param keyColumnMappings  A map defining the column correlations between the current DataFrame and otherDf.
    *                        Keys represent column names in the current DataFrame,
    *                        and values are corresponding column names in otherDf.
    * @param assertion       A function that takes a Double (result of the comparison) and returns a Boolean.
    *                        Defines the condition under which the data in both DataFrames is considered synchronized.
    *                        For example (_ > 0.7) denoting metric value > 0.7 or 70% of records.
+   * @param matchColumnMappings A map defining the column correlations between the current DataFrame and otherDf.
+   *                        These are the columns which we will check for equality, post joining. It's an optional value
+   *                        with defaults to None, which will be derived from `keyColumnMappings` if None.
    * @param hint            Optional. Additional context or information about the synchronization check.
    *                        Helpful for understanding the intent or specifics of the check. Default is None.
    * @return                A [[com.amazon.deequ.checks.Check]] object representing the outcome
-   *                        of the synchronization check. This object can be used in Deequ's verification suite to
+   *                        of the dataset match check. This object can be used in Deequ's verification suite to
    *                        assert data quality constraints.
    *
    */
-  def isDataSynchronized(otherDf: DataFrame, columnMappings: Map[String, String], assertion: Double => Boolean,
-                         hint: Option[String] = None): Check = {
-    val dataSyncAnalyzer = DataSynchronizationAnalyzer(otherDf, columnMappings, assertion)
-    val constraint = AnalysisBasedConstraint[DataSynchronizationState, Double, Double](dataSyncAnalyzer, assertion,
+  def isDatasetMatched(otherDf: DataFrame,
```

Comment:
We could change `otherDf` to `otherDataset`. We use `data` as the name of the parameter for the primary dataset in the Verification suite.